analyzers/tests/SonarAnalyzer.UnitTest/AnalysisContext/SonarAnalysisContextTest.Register.cs

```diff
@@ -56,8 +56,10 @@ public void RegisterNodeAction_UnchangedFiles_SonarAnalysisContext(string unchan
     public void RegisterNodeAction_UnchangedFiles_SonarParametrizedAnalysisContext(string unchangedFileName, bool expected)
     {
         var context = new DummyAnalysisContext(TestContext, unchangedFileName);
-        var sut = new SonarParametrizedAnalysisContext(new(context, DummyMainDescriptor));
+        var sonarContext = new SonarAnalysisContext(context, DummyMainDescriptor);
+        var sut = new SonarParametrizedAnalysisContext(sonarContext);
         sut.RegisterNodeAction<SyntaxKind>(CSharpGeneratedCodeRecognizer.Instance, context.DelegateAction);
+        sonarContext.RegisterCompilationStartAction(c => sut.ExecutePostponedActions(c));
```

Comment:
`SonarParametrizedAnalysisContext.RegisterComplilationStartAction` does not execute the registration right away, for the test to succeed we need to call the `ExecutePostponedActions` as in `ParametrizedDiagnosticAnalyzer`